R/utils-get_code_dependency.R

```diff
@@ -108,25 +110,24 @@ find_call <- function(call_pd, text) {
 #' @noRd
 extract_calls <- function(pd) {
   calls <- lapply(
-    pd[pd$parent == 0, "id"],
+    pd[pd$parent == 0 & pd$token != "COMMENT", "id"],
     function(parent) {
       rbind(
-        pd[pd$id == parent, c("token", "text", "id", "parent")],
+        pd[pd$id == parent, ],
         get_children(pd = pd, parent = parent)
       )
     }
   )
   calls <- Filter(function(call) !(nrow(call) == 1 && call$token == "';'"), calls)
   calls <- Filter(Negate(is.null), calls)
-  calls <- fix_shifted_comments(calls)
```

Comment:
I think this might be needed.

This test is failing.

```r
testthat::test_that("get_code does not break if @linksto is put in the last line", {
  # In some cases R parses comment as a separate expression so the comment is not
  # directly associated with this line of code. This situation occurs when `eval` is in the last
  # line of the code. Other cases are not known but are highly probable.
  code <- c(
    "expr <- quote(x <- x + 1)",
    "x <- 0",
    "eval(expr) #@linksto x"
  )
  q <- eval_code(qenv(), code)
  testthat::expect_identical(
    get_code_g(q, names = "x"),
    code
  )
})
```

Reply:
On it!

Reply:
Connected to https://github.com/insightsengineering/teal.code/pull/216/files#r1827636017

Reply:
Brought back in here 0afef8388f8804df5d6006078edd650bb8bee52a